modules/storage-aws-s3/s3/s3.go

```diff
@@ -102,8 +111,56 @@ func (s *s3) StoreSnapshot(ctx context.Context, snapshot snapshots.Snapshot) err
 	return nil
 }
 
+func (s *s3) makeFilePath(dataPath, relPath string) string {
+	return fmt.Sprintf("%s/%s", dataPath, relPath)
+}
+
+func (s *s3) makeObjectName(snapshotId, srcRelPath string) string {
+	return fmt.Sprintf("%s/%s", snapshotId, srcRelPath)
+}
+
 func (s *s3) RestoreSnapshot(ctx context.Context, snapshotId string) error {
```

Comment:
currently Snapshot entity holds more data than just file paths (like ShardingState and ShardMetadata). I believe those needs to be restored as well. Could you please align with @parkerduckworth to determine what else might be needed for successful restore?

Reply:
These need to be written to disk and read back during the restore

Reply:
I can't find anything more to be restored, resolving this as done, for now